config/seata-config-core/src/main/java/io/seata/config/ConfigurationCache.java

```diff
@@ -106,21 +109,32 @@ public Configuration proxy(Configuration originalConfiguration) {
                 if (method.getName().startsWith(METHOD_PREFIX)
                         && !method.getName().equalsIgnoreCase(METHOD_LATEST_CONFIG)) {
                     String rawDataId = (String)args[0];
-                    ObjectWrapper wrapper = CONFIG_CACHE.get(rawDataId);
                     ObjectWrapper.ConfigType type = ObjectWrapper.getTypeByName(method.getName().substring(METHOD_PREFIX.length()));
                     Object defaultValue = null;
                     if (args.length > 1 && method.getParameterTypes()[1].getSimpleName().equalsIgnoreCase(type.name())) {
                         defaultValue = args[1];
                     }
-                    if (null == wrapper || (null != defaultValue && !Objects.equals(defaultValue, wrapper.lastDefaultValue))) {
-                        Object result = method.invoke(originalConfiguration, args);
-                        // The wrapper.data only exists in the cache when it is not null.
-                        if (result != null) {
-                            wrapper = new ObjectWrapper(result, type, defaultValue);
-                            CONFIG_CACHE.put(rawDataId, wrapper);
+                    final Object finalDefaultValue = defaultValue;
+                    ObjectWrapper wrapper = CollectionUtils.computeIfAbsent(CONFIG_CACHE, rawDataId,
+                        k -> new ObjectWrapper(null, type, finalDefaultValue));
+                    boolean notEquals = null != defaultValue && !Objects.equals(defaultValue, wrapper.lastDefaultValue);
+                    if (wrapper.getData() == null || notEquals) {
+                        synchronized (wrapper) {
+                            if (wrapper.getData() == null || notEquals) {
+                                Object result = method.invoke(originalConfiguration, args);
+                                Object localResult = null;
+                                // When the remote configuration item does not exist, try to read the local configuration item
+                                if (result == null || (defaultValue != null && Objects.equals(result, defaultValue))) {
```

Comment:
1. not only the configuration center, but also the YML configuration file.
2. if yml as configuration file and set as default value, it will read twice.